packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -63,12 +63,12 @@ function _getHashElementScrollMarginTop (selector: string): number {
 }
 
 function _isDifferentRoute (from: RouteLocationNormalized, to: RouteLocationNormalized): boolean {
-  const samePageComponent = to.matched.every((comp, index) => comp.components?.default === from.matched[index]?.components?.default)
+  const isSamePath = to.path === from.path
 
-  if (!samePageComponent) {
+  if (!isSamePath) {
     return true
   }
-  if (samePageComponent && JSON.stringify(from.params) !== JSON.stringify(to.params)) {
+  if (isSamePath && JSON.stringify(from.params) !== JSON.stringify(to.params)) {
```

Comment:
Is this necessary? Since there is !isSamePath check already which returns true it is considered as true otherwise in this case I feel like isSamePath is an unnecessary check and it can be only
`if (JSON.stringify(from.params) !== JSON.stringify(to.params))`

Reply:
oops i only did a rename, didn't really thougt of removing this 😅